components/ai_chat/core/browser/engine/conversation_api_client_unittest.cc

```diff
@@ -217,6 +217,27 @@ TEST_F(ConversationAPIUnitTest, PerformRequest_PremiumHeaders) {
           result.GetDict().Set("queries", std::move(queries));
           data_received_callback.Run(base::ok(std::move(result)));
         }
+        {
+          base::Value result(base::Value::Type::DICT);
+          result.GetDict().Set("type", "webSources");
+          base::Value sources(base::Value::Type::LIST);
+          {
+            base::Value query(base::Value::Type::DICT);
+            query.GetDict().Set("title", "Star Wars");
+            query.GetDict().Set("url", "https://starwars.com");
+            query.GetDict().Set("favicon", "https://starwars.com/favicon");
+            sources.GetList().Append(std::move(query));
+          }
+          {
+            base::Value query(base::Value::Type::DICT);
+            query.GetDict().Set("title", "Star Trek");
+            query.GetDict().Set("url", "https://startrek.com");
+            query.GetDict().Set("favicon", "https://startrek.com/favicon");
+            sources.GetList().Append(std::move(query));
+          }
```

Comment:
this should probably fail for `http://` favicons and `javascript:...` URLs

Reply:
The parsing will succeed but we won't link to the site or download the favicon